syntax = "proto3";

package envoy.extensions.quic.server_preferred_address.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.quic.server_preferred_address.v3";
option java_outer_classname = "BasicServerPreferredAddressConfigProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/quic/server_preferred_address/v3;server_preferred_addressv3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: QUIC server preferred address config]
// [#extension: envoy.quic.server_preferred_address.basic]

// Configuration for BasicServerPreferredAddressConfig.
message BasicServerPreferredAddressConfig {
  // String representation of ipv(4|6) address, i.e. "127.0.0.2".
  // If not specified, none will be configured.
  string ipv4_address = 1;

  string ipv6_address = 2;
}
